discord/voice_client.py

```diff
@@ -107,6 +189,12 @@ def __init__(self, state, timeout, channel):
         self._runner = None
         self._player = None
         self.encoder = opus.Encoder()
+        self.decoders = {}
+        self._ssrc_lookup = {}
+        self._voice_receiver = None
+
+        self._pcm_listeners = []
+        self._speaking_listeners = []
```

Comment:
if you are going to split by userid/ssrc this should be a mapping of user id (or ssrc) to listener (or list of listeners). not a list of listeners that then filter (wasted cpu time)

Reply:
on an architectural level, you could have this be a mapping of user id (or ssrc) to an object that handles the decoding (it could also own the decoder as well). 

You should also add some logic such that if the ssrc for a user changes, the decoder is reset as well.